src/psyclone/domain/gocean/transformations/gocean_const_loop_bounds_trans.py

```diff
@@ -169,26 +229,18 @@ def apply(self, node, options=None):
         # Fortran reader needed to parse constructed bound expressions
         fortran_reader = FortranReader()
 
-        const = GOceanConstants()
         # Update all GOLoop bounds with the new variables
         for loop in node.walk(GOLoop):
-            # Look for a child kernel in order to get the index offset.
-            index_offset = loop.index_offset
-
-            if index_offset not in const.SUPPORTED_OFFSETS:
-                raise TransformationError(
-                    "Constant bounds generation not implemented for a grid "
-                    "offset of '{0}'. Supported offsets are {1}"
-                    "".format(index_offset, const.SUPPORTED_OFFSETS))
 
             # Chose the variable depending if it is an inner or outer loop
             if loop.loop_type == "inner":
                 stop = i_stop.name
-            else:
+            elif loop.loop_type == "outer":
                 stop = j_stop.name
+            # the validate method raises and error for any other loop_type
```

Comment:
True but we should still be defensive and raise an InternalError here (since "it should never happen").

Reply:
Done